nautobot/core/jobs/cleanup.py

```diff
@@ -48,6 +49,23 @@ class Meta:
         description = "Delete ObjectChange and/or JobResult/JobLogEntry records older than a specified cutoff."
         has_sensitive_variables = False
 
+    def recursive_delete_with_cascade(self, queryset):
+        """
+        Recursively deletes all related objects with CASCADE for a given queryset.
+
+        Args:
+            queryset (QuerySet): The queryset of objects to delete.
+        """
+        related_objects = queryset.model._meta.related_objects
+
+        for related_object in related_objects:
+            if related_object.on_delete == CASCADE:
+                related_model = related_object.related_model
+                related_field_name = related_object.field.name
+                cascade_queryset = related_model.objects.filter(**{f"{related_field_name}__in": queryset})
+                self.recursive_delete_with_cascade(cascade_queryset)
+        return queryset.delete()
```

Comment:
Would be nice to capture and aggregate the return values from each `delete` call and report that in the job logs. E.g.:

```
Deleted 12234 related JobLogEntry records
Deleted 1232130 related SyncLogEntry records
Deleted 22 related Sync records
Deleted 57 JobResult records
```